app/controllers/applicants_controller.rb

```diff
@@ -1,34 +1,7 @@
 class ApplicantsController < ApplicationController
-  resource_description do
-    description <<-END_OF_TEXT
-    == Description
-      This endpoint will create an Applicant and associate it with
-      an existing Assessment which has been created with:
-
-        POST /assessments
-
-    END_OF_TEXT
-  end
-  api :POST, "assessments/:assessment_id/applicant", "Create Applicant and attach it to an existing Assessment (create assessment first with POST /assessments)"
-  formats(%w[json])
-  param :assessment_id, :uuid, required: true, desc: "The assessment id to which this applicant relates - must have been created prior to this call with POST /assessments"
-  param :applicant, Hash, desc: "Describes basic info about the applicant", required: true do
-    param :date_of_birth, Date, date_option: :today_or_older, required: true, desc: "The applicant's date of birth"
-    param :involvement_type, Applicant.involvement_types.values, required: true, desc: "How the applicant is involved in the case"
-    param :has_partner_opponent, :boolean, required: true, desc: "Whether or not the applicant's partner is an opponent in the case"
-    param :receives_qualifying_benefit, :boolean, required: true, desc: "Whether or not the applicant receives a qualifying benefit"
-  end
-
-  returns code: :ok, desc: "Successful response" do
-    property :success, %w[true], desc: "Success flag shows true"
-    property :errors, [], desc: "Empty array of error messages"
-  end
-  returns code: :unprocessable_entity do
-    property :success, %w[false], desc: "Success flag shows false"
-    property :errors, array_of: String, desc: "Description of why object invalid"
-  end
-
   def create
+    return render_unprocessable(json_validator.errors) unless json_validator.valid?
```

Comment:
ideally this should be internal to the creation service actually?

Reply:
@jsugarman I am not sure, I intentionally added return render_unprocessable(json_validator.errors) unless json_validator.valid? to stop it hitting the creation service if the validation failed (it previously didn’t hit the creation service if the apipie validation failed) - is that not a valid approach? (To me we are using json_schema to validate the request params so it makes sense to call it from the controller. We pass creation_service a hash input so we would probably have to turn this back into json in order to validate it with json_schema if we did that in the creation service). Happy to pair if you want to?

Reply:
@jsugarman I have added a commit to move the call to json_validator from the controller to the creator.